index.js

```diff
@@ -3983,6 +3983,31 @@
   S.splitOn =
   def('splitOn', {}, [$.String, $.String, $.Array($.String)], splitOn);
 
+  //# splitOnRegex :: RegExp -> String -> Array String
+  //.
+  //. Type-checked version of String#split that accepts RegEx as a separator.
+  //.
+  //. Takes a pattern and a string, and returns the result of splitting the
+  //. string at every non-overlapping occurrence of the pattern (possibly
+  //. removing the matched substrings, depending on a pattern).
```

Comment:
> possibly removing the matched substrings

It's simpler to state that the function *always* replaces each match with a (possibly empty) splice containing the captured substrings. If the pattern is `/(-)/`, say, it will *appear* that the matches are not removed:

```javascript
> 'foo-bar-baz'.split(/(-)/)
['foo', '-', 'bar', '-', 'baz']
```

We just happen to be replacing `'-'` with the splice `['-']`.

Changing the number of capturing groups is instructive:

```javascript
> 'foo-bar-baz'.split(/-/)
['foo', 'bar', 'baz']

> 'foo-bar-baz'.split(/(((-)))/)
['foo', '-', '-', '-', 'bar', '-', '-', '-', 'baz']
```

I think we should remove the parenthesized clause and rely on MDN to describe the treatment of capturing groups.